naxsi_src/naxsi_runtime.c

```diff
@@ -1133,9 +1133,11 @@ naxsi_create_log_array(ngx_http_request_ctx_t* ctx,
   sz_left -= sub;
   offset += sub;
 
-  sub = NAXSI_REQUEST_ID_SIZE << 1;
+  const u_char* req_id = naxsi_request_id(r);
+
+  sub = strlen((const char*)req_id);
   if (sz_left > (100 + sub)) {
-    ngx_hex_dump(fragment->data + offset, ctx->request_id, NAXSI_REQUEST_ID_SIZE);
+    strcpy((char*)(fragment->data + offset), (const char*)req_id);
```

Comment:
use `strncpy`, with size `sz_left - offset`

Reply:
For what?
**naxsi_request_id()** always returns either an empty string or **ctx->request_id** -- an array of size NAXSI_REQUEST_ID_SIZE+1, with the last byte **always** 0 (calloc when allocated)

And if we really want to play it safe, then not sz_left, but NAXSI_REQUEST_ID_SIZE :smile: 

It would be necessary to rewrite the logging algorithm more carefully. There are things that are wrong

Reply:
yeah, the logging algo is a remaining that needs to be rewritten more efficiently and safe.